Source/Scene/ShadowMap.js

```diff
@@ -64,33 +83,70 @@ define([
         OrthographicFrustum,
         Pass,
         PerInstanceColorAppearance,
+        PerspectiveFrustum,
         Primitive) {
     'use strict';
 
-    // TODO : doc
     /**
      * @private
      */
-    function ShadowMap(context, camera) {
+    function ShadowMap(context, lightCamera) {
         this.enabled = false;
 
+        // Uniforms
+        this._shadowMapMatrix = new Matrix4();
+        this._shadowMapTexture = undefined;
+
+        this._framebuffer = undefined;
+        this._shadowMapSize = 1024;
+
+        this._lightCamera = lightCamera;
+        this._shadowMapCamera = new ShadowMapCamera();
+        this._sceneCamera = undefined;
+        this._farPlane = 100.0; // Limit the far plane of the scene camera
+        this._fitToScene = true;
+
+        var numberOfCascades = 4;
+        this._cascadesEnabled = true;
+        this._numberOfCascades = numberOfCascades;
+
+        this._cascadeCameras = new Array(numberOfCascades);
+        this._cascadePassStates = new Array(numberOfCascades);
+        this._cascadeViewports = new Array(numberOfCascades);
+
+        // Uniforms
+        this._cascadeSplitsNear = new Cartesian4();
+        this._cascadeSplitsFar = new Cartesian4();
+        this._cascadeOffsets = new Array(numberOfCascades);
+        this._cascadeScales = new Array(numberOfCascades);
+
+        for (var i = 0; i < numberOfCascades; ++i) {
```

Comment:
Does it make sense to factor all this into a new separate `ShadowMapCascade` class?